### PR TITLE
Add integration to MessageHub

### PR DESCRIPTION
**Pull request by thbaan**
## References

* #0000

### CODE REVIEW

File: source/Energinet.DataHub.MeteringPoints.EntryPoints.Outbox/local.settings.sample.json [hunk 1]
```diff
@@ -1,16 +1,18 @@
...
     "METERING_POINT_CREATED_TOPIC": "<Topic name for MeteringPointCreated events>",
     "METERING_POINT_CONNECTED_TOPIC": "<Topic name for MeteringPointConnected events>",
     "METERINGPOINT_DB_CONNECTION_STRING": "<Main SQL database connection string>",
     "EVENT_MESSAGE_DISPATCH_TRIGGER_TIMER": "*/10 * * * * *",
-    "ACTOR_MESSAGE_DISPATCH_TRIGGER_TIMER": "*/10 * * * * *",
-    "TEMP_POST_OFFICE_CONNECTION_STRING": "",
-    "TEMP_POST_OFFICE_SHARE": ""
+    "ACTOR_MESSAGE_DISPATCH_TRIGGER_TIMER": "*/10 * * * * *"
+  },
+  "ConnectionStrings": {
```
Comment (Korgath) [inline source/Energinet.DataHub.MeteringPoints.EntryPoints.Outbox/local.settings.sample.json:14]: Consider moving all connection strings here?

File: source/Energinet.DataHub.MeteringPoints.EntryPoints.SubPostOffice/local.settings.sample.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  "IsEncrypted": false,
+  "Values": {
+    "AzureWebJobsStorage": "UseDevelopmentStorage=true",
+    "FUNCTIONS_WORKER_RUNTIME": "dotnet-isolated",
+    "APPINSIGHTS_INSTRUMENTATIONKEY": "<Azure application insights instrumentation key>",
+    "METERINGPOINT_DB_CONNECTION_STRING": "<Main SQL server database connection string>",
+    "METERINGPOINT_QUEUE_TOPIC_NAME": "<Name of Service Bus queue for receiving business requests and internal commands>",
+    "METERINGPOINT_QUEUE_CONNECTION_STRING": "<Service Bus connection string where business requests and internal commands are received>",
+  },
+  "ConnectionStrings": {
```
Comment (Korgath) [inline source/Energinet.DataHub.MeteringPoints.EntryPoints.SubPostOffice/local.settings.sample.json:11]: Same as above

**Review by Korgath — APPROVED** [review]

File: source/Energinet.DataHub.MeteringPoints.EntryPoints.SubPostOffice/Functions/BundleDequeuedQueueSubscriber.cs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+using System.Threading.Tasks;
+using Energinet.DataHub.MeteringPoints.Infrastructure.Correlation;
+using Energinet.DataHub.MeteringPoints.Infrastructure.SubPostOffice;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Extensions.Logging;
+
+namespace Energinet.DataHub.MeteringPoints.EntryPoints.SubPostOffice.Functions
+{
+    public class BundleDequeuedQueueSubscriber
```
Comment (kft) [inline source/Energinet.DataHub.MeteringPoints.EntryPoints.SubPostOffice/Functions/BundleDequeuedQueueSubscriber.cs:23]: It is possible to use more intention revealing naming ? The purpose of this is a bit unclear to me

File: source/Energinet.DataHub.MeteringPoints.Infrastructure/EDI/DocumentType.cs [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+namespace Energinet.DataHub.MeteringPoints.Infrastructure.EDI
+{
+    public sealed class DocumentType : EnumerationType
+    {
+        public static readonly DocumentType CreateMeteringPointAccepted = new(1, nameof(CreateMeteringPointAccepted), typeof(ConfirmMessage));
+        public static readonly DocumentType CreateMeteringPointRejected = new(2, nameof(CreateMeteringPointRejected), typeof(RejectMessage));
+        public static readonly DocumentType ConnectMeteringPointAccepted = new(3, nameof(ConnectMeteringPointAccepted), typeof(ConfirmMessage));
+        public static readonly DocumentType ConnectMeteringPointRejected = new(4, nameof(ConnectMeteringPointRejected), typeof(RejectMessage));
+        public static readonly DocumentType AccountingPointCharacteristicsMessage = new(5, nameof(AccountingPointCharacteristicsMessage), typeof(AccountingPointCharacteristicsMessage));
+
```
Comment (kft) [inline source/Energinet.DataHub.MeteringPoints.Infrastructure/EDI/DocumentType.cs:29]: Can this be done more generic ? Every time a business process is implemented, we have to change this class. A violation of OCP.

File: source/Energinet.DataHub.MeteringPoints.Infrastructure/SubPostOffice/ISubPostOfficeClient.cs [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        /// Request bundle from Local Post Office
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing the result of the asynchronous operation.</returns>
+        Task CreateBundleAsync(byte[] request, string sessionId);
+
+        /// <summary>
+        /// Notify that bundle has been dequeued
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing the result of the asynchronous operation.</returns>
+        Task BundleDequeuedAsync(byte[] notification);
```
Comment (kft) [inline source/Energinet.DataHub.MeteringPoints.Infrastructure/SubPostOffice/ISubPostOfficeClient.cs:34]: Is this some kind of a callback ?

File: source/Energinet.DataHub.MeteringPoints.Infrastructure/SubPostOffice/PostOfficeMessageBlobFactory.cs [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+// namespace Energinet.DataHub.MeteringPoints.Infrastructure.SubPostOffice
+// {
+//     public static class PostOfficeMessageBlobFactory
+//     {
+//         public static PostOfficeMessageBlob Create(string blobName, string content)
+//         {
+//             return new PostOfficeMessageBlob(blobName, new BinaryData(content));
+//         }
+//     }
+// }
```
Comment (kft) [inline source/Energinet.DataHub.MeteringPoints.Infrastructure/SubPostOffice/PostOfficeMessageBlobFactory.cs:26]: Commented out code

File: source/Energinet.DataHub.MeteringPoints.Infrastructure/SubPostOffice/SubPostOfficeClient.cs [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+        private readonly IPostOfficeMessageMetadataRepository _postOfficeMessageMetadataRepository;
+
+        public SubPostOfficeClient(
+            IStorageHandler postOfficeStorageHandler,
+            IPostOfficeMessageMetadataRepository postOfficeMessageMetadataRepository,
+            IMessageDispatcher messageDispatcher,
+            IDataBundleResponseSender dataBundleResponseSender,
+            IDequeueNotificationParser dequeueNotificationParser,
+            IRequestBundleParser requestBundleParser,
+            IBundleCreator bundleCreator)
```
Comment (kft) [inline source/Energinet.DataHub.MeteringPoints.Infrastructure/SubPostOffice/SubPostOfficeClient.cs:47]: There are many dependencies here. Maybe this class could be split ?

**Review by kft — APPROVED** [review]
A couple of comment. Other than that LGTM.

File: source/Energinet.DataHub.MeteringPoints.Infrastructure/SubPostOffice/SubPostOfficeClient.cs (hunk 7 above)
Comment (thbaan) [reply source/Energinet.DataHub.MeteringPoints.Infrastructure/SubPostOffice/SubPostOfficeClient.cs:47]: Good point. In fact - "Data Available" has already been taken out of this general client. Further splitting can be done at a later point.

File: source/Energinet.DataHub.MeteringPoints.EntryPoints.SubPostOffice/Functions/BundleDequeuedQueueSubscriber.cs (hunk 3 above)
Comment (thbaan) [reply source/Energinet.DataHub.MeteringPoints.EntryPoints.SubPostOffice/Functions/BundleDequeuedQueueSubscriber.cs:23]: An event will be raised in MessageHub when an actor dequeues a message. This entry point is a service bus trigger to receive that event.

**Pull request merged**